src/services/__tests__/token.service.unit.spec.js

```diff
@@ -0,0 +1,223 @@
+import { ROLES, tokenService } from '../token.service.js'
+import {
+  getMockResponseToolkit,
+  getServerDetails
+} from '../../test-utils/server-test-utils.js'
+import axios from 'axios'
+import { getSystemUserByOid } from '../system-users.service.js'
+import jwksClient from 'jwks-rsa'
+import jwt from 'jsonwebtoken'
+
+jest.mock('axios')
+jest.mock('jsonwebtoken')
+jest.mock('jwks-rsa')
+jest.mock('../system-users.service.js')
+
+describe('token.service.unit', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+
+    axios.get.mockResolvedValue({
+      data: {
+        jwks_uri: 'https://example.com/jwks'
+      }
+    })
+
+    process.env.OIDC_WELL_KNOWN_URL =
+      'https://example.com/.well-known/openid-configuration'
+    process.env.OIDC_CLIENT_ID = 'test-client-id'
+  })
+
+  const getMockResponseToolkitTakeover = () => {
+    const responseObject = {
+      code: jest.fn().mockReturnThis(),
+      takeover: jest.fn().mockReturnThis(),
+      payload: null,
+      statusCode: null
+    }
+
+    const response = jest.fn().mockImplementation((payload) => {
+      responseObject.payload = payload
+      return responseObject
+    })
+
+    responseObject.code.mockImplementation((statusCode) => {
+      responseObject.statusCode = statusCode
+      return responseObject
+    })
+
+    return {
+      response,
+      continue: 'continue-response'
+    }
+  }
+
+  describe('tokenService', () => {
+    it('should continue if no token is provided', async () => {
+      const h = getMockResponseToolkit()
+      const result = await tokenService(
+        getServerDetails({ headers: { token: undefined } }),
+        h
+      )
+      expect(result).toBe(h.continue)
+    })
+
+    it('should return 401 and an error message if token header is invalid', async () => {
+      jwt.decode.mockReturnValue(null)
+      const result = await tokenService(
+        getServerDetails({ headers: { token: 'abc123' } }),
+        getMockResponseToolkitTakeover()
+      )
+
+      expect(result.statusCode).toBe(401)
+      expect(result.payload).toStrictEqual({ error: 'Invalid token header' })
+    })
+
+    it('should return 401 if token header has no kid', async () => {
+      jwt.decode.mockReturnValue({ header: {} })
+
+      const result = await tokenService(
+        getServerDetails({ headers: { token: 'abc123' } }),
+        getMockResponseToolkitTakeover()
+      )
+
+      expect(result.statusCode).toBe(401)
+      expect(result.payload).toStrictEqual({ error: 'Invalid token header' })
+    })
```

Comment:
this and the previous test are identical other than the return value of the decode mock, so should be combined using it.each